src/main/java/com/microsoft/azure/vmagent/builders/AdvancedImageBuilder.java

```diff
@@ -12,7 +12,8 @@ public AdvancedImageBuilder(AdvancedImageFluent<?> fluent) {
     public AdvancedImageBuilder(AdvancedImageFluent<?> fluent, AdvancedImage image) {
         this.fluent = fluent;
         fluent.withCustomImage(image.getImage());
-        fluent.withReferenceImage(image.getImagePublisher(),
+        fluent.withReferenceImage(image.getImageId(),
```

Comment:
Seems here is a left bug. We should justify the `imageReferenceType` first and then use ` fluent.withCustomImage()`, `fluent.withReferenceImage()` or `withCustomManagedImage()`